src/frontend/src/lib/components/pages/SettingsSignedIn.svelte

```diff
@@ -10,12 +10,25 @@
 	import { i18n } from '$lib/stores/i18n.store';
 	import { replaceOisyPlaceholders } from '$lib/utils/i18n.utils';
 	import TokensZeroBalanceToggle from '$lib/components/tokens/TokensZeroBalanceToggle.svelte';
+	import { POUH_ENABLED } from '$lib/constants/app.constants';
+	import { userHasPouhCredential } from '$lib/derived/has-pouh-credential';
+	import { requestPouhCredential } from '$lib/services/request-pouh-credential.services';
+	import { fade } from 'svelte/transition';
+	import { busy } from '$lib/stores/busy.store';
 
 	let remainingTimeMilliseconds: number | undefined;
 	$: remainingTimeMilliseconds = $authRemainingTimeStore;
 
 	let principal: Principal | undefined | null;
 	$: principal = $authStore?.identity?.getPrincipal();
+
+	const getPouhCredential = async () => {
+		if (nonNullish(principal)) {
+			busy.show();
+			await requestPouhCredential({ credentialSubject: principal });
```

Comment:
you do not need a `try finally` here to ensure `busy.stop()` is called in case of errors as well

Reply:
Good point, done!